src/librustc_typeck/collect.rs

```diff
@@ -2603,6 +2603,10 @@ fn codegen_fn_attrs(tcx: TyCtxt<'_>, id: DefId) -> CodegenFnAttrs {
             }
         } else if attr.check_name(sym::link_name) {
             codegen_fn_attrs.link_name = attr.value_str();
+        } else if attr.check_name(sym::link_ordinal) {
+            if let Some(ordinal) = check_link_ordinal(tcx, attr) {
```

Comment:
```suggestion
            if let ordinal@Some(_) = check_link_ordinal(tcx, attr) {
```

Reply:
Didn't know i can do this in `if let` too! Thank you

Reply:
You can do it anywhere you can use a pattern. Rust is compositional that way. ;)